### DISCUSSION TITLE
Are there cameras supported by INDI without pre built binary libraries?

### DISCUSSION THREAD

**Discussion by mattiaverga** [indilib/indi#2080] [General]
As I'm about to buy my first planetary camera, can you tell me what cameras (if any) are fully supported by INDI without requiring pre built binary blobs?
I'm maintaining INDI in Fedora linux and explored the possibility to package some of the SDKs distributed within INDI, but that doesn't seem  possible. The only cameras I see without binary blobs are apogee and fli, but they're obviously way over my budget...

BTW, while looking at packaging the Player One libraries, another Fedora user has pointed out that PLayer One SDK license only mention the "use" case, but it doesn't allow redistribution... so, legally speaking, INDI would not be able to redistribute the SDK in the source code.

**Comment by bluthen** [comment]
> You can use our company's products and this SDK to develop any products without any restrictions.

Guess it depends on what they mean by the above line. How can you develop any product without any restrictions if you can't distribute it?

**Comment by mattiaverga** [comment]
I guess the modifications to the standard MIT license wording are strange. The standard MIT license text explicitly says:
```
Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so
```
while Player One license only explicitly list the "use" case.
I'm not a lawyer, but I know the legal Red Hat team, which supervises such license oddities in Fedora, would object to those modifications because explicit words are better than implicit meaning or intentions...

However, this was just to raise your attention on that (maybe you can ask Player One to clarify their license text by using a standard text) as I doubt I will be able to package those binary blobs in Fedora.

**Comment by aaronwmorris** [comment]
The short answer is "no".  The real 600lbs gorilla here is probably Sony who manufactures of almost all modern CMOS sensors in use today.  The SDK might have a license, the blob is probably hindered by 1000 patents and licenses owned by Sony and a dozen other companies.  No Linux distribution is going to touch the 3rd party repository with a 10 foot pole, which is probably why the 3rd party repository exists in the first place.

Most astronomy software that distributes camera drivers likely operates in a legal gray area.

**Comment by bluthen** [comment]
I think the DSLRs using gphoto don't need blobs. I believe starlight cameras don't use blobs. Maybe some "webcams"
